### DISCUSSION TITLE
Looking for Ideas to Improve the 3D Printed Enclosure Mounting

### DISCUSSION THREAD

**Discussion by keenanjohnson** [Ribbit-Network/ribbit-network-frog-hardware#89] [General]
As I've started to put together more than a few of the 3D printed enclosures, I've become aware that tightening the bolts attaching the main box enclosure to the top sensor shroud and head is a bit tedious. See images below.

**I think there are two things making this hard:**
* You have to exert a bit of force into the bolt to get it to "bite" into the 3D Printed Hole on the other side.
* There isn't much room to turn a standard hex wrench inside the enclosure. At best you can probably get just over a quarter turn which means a lot of turning motion haha.

**Ideas for Solution**
I have a few ideas for how to improve this, but I would love to hear other opinions!

* Add a heat insert into the tapped hole - [I've thought about adding something like this into the hole. 
](https://www.mcmaster.com/heat-set-inserts/). This adds a new part to the BOM which isn't ideal, but it would significantly decrease the force needed on the bolt to get it to thread properly
* Replace bolt with Clips - [The previous off-the-shelf enclosure](https://www.amazon.com/AcuRite-06054M-Temperature-Humidity-Radiation/dp/B01M64ISDE/ref=sr_1_14?keywords=accuweather+enclosure&qid=1639617446&sr=8-14) we used to use is held together entirely by plastic clips molded into the shroud layers. Perhaps we could switch the bolts to 3D printed clips? This might make the shroud harder to disasemble, but doesn't add any parts needed to build a frog

![IMG_0732](https://user-images.githubusercontent.com/2559382/146289526-b16af4d1-d925-4411-b40f-e7797cf6dd07.JPG)
![IMG_0730](https://user-images.githubusercontent.com/2559382/146289540-a5f57a14-2e91-4238-8017-6a59fcdacd06.JPG)

**Comment by keenanjohnson** [comment]
@eaudiffred perhaps you might have opinions here?

**Comment by eaudiffred** [comment]
@keenanjohnson Two very good points.  
1. For some reason OnShape does not model the threads.  It's currently just modeling a tap hole for the bolt size that we select.  With some other CAD programs I've been able to print the threads.  Works pretty good down to M4 size.  I'll see if there is an option in OnShape to actually model the threads so that they get printed.  
    a. The threaded insert is a good idea, but I agree more parts is a pain and the negative out weights the positive.  
    b. I've been tapping the holes for the couple of enclosures I've made and its significantly easier to get the bolts in.  They just spin right into place.  Can't assume that everyone has an M4 tap laying around though, so we will figure out a way around that.

2. For this I would recommend installing the board after the screws.  Without the board in there, you can use the long side of the hex wrench and just spin the bolt into place (if the threads are already cut).
     a. We could angle the bolts slightly more toward the center to get the wrench away from the wall.  I could also try and get a compound angle on the bolts so that the end of the wrench also sticks up away from the enclosure.  
     b. I like the clip idea too.  Just have to think about it a little more and figure out a design that is printable.  I'll check out the Acurite design.  Maybe we can copy it.  We might be able to print a clip piece that pops into the body and the two shrouds.

**Comment by eaudiffred** [comment]
Instead of the threaded insert, we could make a slot to slide in a nut.  Still adds parts, but the nuts will be cheaper than the insert.  I don't love that idea because it makes assembly just a little harder, but it can be added to the potential solution list.

**Comment by keenanjohnson** [comment]
Yeah the captive nut is another good idea! 

Here's what the Accuweather Injection-molded clips look like. I'm not sure if this geometry is feasible to 3D print or not.

![IMG_0756](https://user-images.githubusercontent.com/2559382/146441191-b983a98e-d574-447e-915e-e67afcd4df55.JPG)
![IMG_0755](https://user-images.githubusercontent.com/2559382/146441202-14052655-2f9c-40c4-a619-c220f7a86fb7.JPG)
![IMG_0754](https://user-images.githubusercontent.com/2559382/146441208-3f362744-9639-45fe-8872-15f68c21a524.JPG)
![IMG_0753](https://user-images.githubusercontent.com/2559382/146441213-630e0c7e-91f0-4eb3-9ecd-269cf0aeb809.JPG)

**Comment by eaudiffred** [comment]
What do you think about this?  Only thing I'm worried about is how securely it holds together.  

I think I'd want to test the fit and function of this first.  If successful, then figure out how to design a clip between the lower shroud and head piece. 

![image](https://user-images.githubusercontent.com/88562047/147031239-75d211cd-7596-471c-86b2-4c2a85ea25f0.png)
![image](https://user-images.githubusercontent.com/88562047/147031276-13fb3e23-e016-4b8a-aa83-46ee1a0bf88d.png)
![image](https://user-images.githubusercontent.com/88562047/147031300-45e5105b-b5e1-4e12-8ad2-ea26aa4a5448.png)
![image](https://user-images.githubusercontent.com/88562047/147031337-35268924-d5c8-4bae-82da-adf877258459.png)

**Reply by keenanjohnson** [reply]
Oh this looks promising to me! The PETG plastic is quite springy, so I'm not worried about it breaking down over time I think. 

Have you already started a test print?

**Reply by keenanjohnson** [reply]
I can if you have not :)

**Reply by eaudiffred** [reply]
I have not started one. Wont be able to until I get back in town on the 30th.

**Comment by eaudiffred** [comment]
I just finished printing a test.  Works great!!  Much better than I hoped it would.  Its snug, so not supper easy to snap into place, but it fits tight and feels rigid.  I'll start thinking of a way to incorporate the head.  (Ignore the poor print quality.  I tried to print it fast and had bad bed adhesion and lots of warping)
![16411429464171734581804422248649](https://user-images.githubusercontent.com/88562047/147883390-b2b7aedd-0f17-4e1b-978c-7e20a5141958.jpg)
![16411429926747591349966785340799](https://user-images.githubusercontent.com/88562047/147883406-74d14975-b1fe-4cf7-beca-aa641dd94e7b.jpg)
![16411430214042169181639941522356](https://user-images.githubusercontent.com/88562047/147883415-e45f59c3-21f2-4afe-a7e0-f8486c75a7ee.jpg)

**Reply by keenanjohnson** [reply]
Wow this is great! This will be really helpful I think for easing the assembly for say a student and reduces our parts required! Wins all around!

**Comment by eaudiffred** [comment]
Are all of the electronics on the top of the housing up-to-date?  
![image](https://user-images.githubusercontent.com/88562047/148096492-05195215-2a63-4aac-b167-5098379bb2b6.png)

**Comment by eaudiffred** [comment]
Depending on the available space, I'm thinking this might be a solution:

![image](https://user-images.githubusercontent.com/88562047/148097075-64afac0f-48d8-49e8-a8ea-92ca8b61e6d8.png)

**Comment by keenanjohnson** [comment]
Oh good call! The electronics are not. I've made the final decision today to switch to a different GPS part that doesn't include the incredibly huge USB cable. #91.

We'll need to update the CAD, but the GPS unit will become much smaller and has a very smaller connector. (Product Dimensions: 25.5mm x 25.4mm x 8.2mm). https://www.adafruit.com/product/4415

**Comment by keenanjohnson** [comment]
I can update that cad today @eaudiffred

**Comment by eaudiffred** [comment]
Ok, cool.  I think I'll have to claim some of that space for the frog head clips, but it looks like with the smaller GPS there will still be plenty of space available.

**Comment by keenanjohnson** [comment]
Yes here you can see how it looks in the built assembly, so this is a great space saver

![image](https://user-images.githubusercontent.com/2559382/148098657-a65bbf6e-2720-41a5-91c9-a17a754bd753.png)

Should I modify this in your clip assembly branch and then we can merge everything in together later?

**Reply by eaudiffred** [reply]
Sure, that sounds like a good plan.

**Comment by keenanjohnson** [comment]
Done! 
![image](https://user-images.githubusercontent.com/2559382/148100004-25559a90-30f5-4545-8c9f-c858137aaf72.png)

**Comment by eaudiffred** [comment]
I made some progress on design yesterday. 
![image](https://user-images.githubusercontent.com/88562047/148398116-a2729a39-a38b-4bb1-aa14-295febc50c70.png)

 One thing I'm worried about with this design is disassembly.  Once the head is on, it seems like it'll be difficult to remove.  One way to fix that would be to modify the clips to slip in and out both directions.  That might make it little less secure/rigid.  Any thoughts?

![image](https://user-images.githubusercontent.com/88562047/148398446-be591780-0096-41bf-8695-69b495216f21.png)

**Comment by keenanjohnson** [comment]
It makes a lot of sense to me that making it easy to disassemble would be important! So I would say let's try it!

**Comment by keenanjohnson** [comment]
Ok @eaudiffred ! I'm back with my printers now, is this ready for another test print on my side you think?

**Comment by eaudiffred** [comment]
Yes!  I think it's ready to go.  I was planning to do a test this week too.

**Comment by keenanjohnson** [comment]
Awesome! I can start one today if that's helpful.

**Reply by eaudiffred** [reply]
Defiantly!

**Comment by keenanjohnson** [comment]
How did you handle the supports for the head top piece now that there are clips coming down?

**Reply by eaudiffred** [reply]
Good Question!  I have not dumped it into Cura to experiment yet.  I'm thinking that the Tree supports will work similar to how the last version was printed.  The clips hang down further, so less support underneath them.  The top/underside of the head will need plenty of support though.  That's where it might get tricky.  

I'll have some time this evening, I think, to mess around with it.  If you start one today, let me know what you try.

**Reply by keenanjohnson** [reply]
Sounds good! I started a print yesterday, but it failed due to a poor bed leveling on my part I think. I realized after it failed that I probably didn't think about the supports enough. I had it set on the tree supports, but I want to take another look at it today and try again.

**Comment by keenanjohnson** [comment]
I decided actually to try out a different slicer program called [PrusaSlicer](https://www.prusa3d.com/page/prusaslicer_424/https://www.prusa3d.com/page/prusaslicer_424/), which is another open source slicer I've heard can work better for Ender printers on forums.

I'm trying the print with this kind of support which seems pretty minimum! We'll see if it works haha

![image](https://user-images.githubusercontent.com/2559382/149403301-8b9aa236-4bad-4dd1-84db-45b9df32ecf7.png)

**Comment by eaudiffred** [comment]
Cool!  How did it turn out?

This is the first option I'd like to try in Cura.  Its the option requiring the least amount of time and material, so it may fail miserably, but I can add material and structure from here.  

I need to level my printer again and re-zero the head.  Hopefully I can get a print started tomorrow.  

![image](https://user-images.githubusercontent.com/88562047/149434001-64b29d0a-1927-46bd-ab86-5e400f4a7ae4.png)

**Reply by keenanjohnson** [reply]
Wow my print turned out great. I was unsure if the petg could bridge that full gap but it seemed to do fine!

![image](https://user-images.githubusercontent.com/2559382/149568053-c0be7eda-32d1-4250-9bb6-f06fdb17ddf4.jpeg)

![image](https://user-images.githubusercontent.com/2559382/149568077-494aa725-e3c7-44ab-9e4a-fdde3e5012b1.jpeg)

![image](https://user-images.githubusercontent.com/2559382/149568103-30b3e0bf-9ea3-4d6b-980a-60cfe6f1179b.jpeg)

![image](https://user-images.githubusercontent.com/2559382/149568118-f1c1fd43-0c81-4d14-a553-b8028167f044.jpeg)

![image](https://user-images.githubusercontent.com/2559382/149568128-4a9e3bf2-955e-468a-a7c1-4144334ca0ed.jpeg)

**Reply by eaudiffred** [reply]
Wow!  I'm really surprised it could bridge that gap too!  Can you post a screen shot of the settings you used?

Is it snug/secure when clipped together?

**Comment by keenanjohnson** [comment]
My settings were these:

* Bed Temp: 85 C
* Nozzle Temp: 250
* Layer Height: .2 mm
* Infill 15%
* Default supports, build plate only

I used the default Ender 3 V2 Profile in Prusa Slicer.

It feels fairly snug when I clipped it together. I've found that I can definitely pull it apart, but the force required feels about right to me that it's not loose, but also not too hard to separate.

**Comment by keenanjohnson** [comment]
![image](https://user-images.githubusercontent.com/2559382/149576988-5e82729e-1942-44fd-836c-f940f554c063.png)
![image](https://user-images.githubusercontent.com/2559382/149577017-55d83edf-0613-47e6-b0ce-c16fb211cb73.png)

**Comment by eaudiffred** [comment]
I printed off a lower shroud and a head over the weekend.  Seems like I need to slightly widen the slot in the lower shroud so that the clips fit in a little easier.  Other than that it seems to fit well.  

Any other thoughts or changes?

**Reply by keenanjohnson** [reply]
Great idea! Yeah the alignment was a bit tricky on my second one, so widening it should help. Other than that, no I think we are good! I'm going to be building the first full sensor with clips today and starting build instructions so I think we're good to merge that branch into the main branch in OnShape and do any CAD cleanup work that needs doing :).

**Reply by keenanjohnson** [reply]
Let me know when you've made your clip changes and I'll print another here and commit my PrusaSlicer files.

**Reply by eaudiffred** [reply]
Changes are made.  Made it 1mm wider so that there is 0.5mm clearance on either side.

**Reply by eaudiffred** [reply]
I right clicked the hollow blue circle while the latest (purple) version was highlighted.  Is selecting "Merge into current workspace" the correct way to do this?  Just want to make sure before I screw something up, lol.  
![image](https://user-images.githubusercontent.com/88562047/150207687-09e4e9e1-2ad4-455e-8299-fc8b4f52d21f.png)

**Reply by keenanjohnson** [reply]
Yes! That's correct if you are currently viewing the Main branch, then clicking that will merge it into main :)

**Reply by eaudiffred** [reply]
Ok, will you double check me on that.  Nothing seemed to happen, but it did add the merge in the change history.

**Reply by keenanjohnson** [reply]
Yeah that looks good to me! I can see your changes in the main branch with the clearance. You can delete the old branch once you are ready to eliminate confusion in the userinterface as well.

![image](https://user-images.githubusercontent.com/2559382/150229804-a8d059cc-4f93-4663-bb76-7ed50bcafb18.png)

**Comment by eaudiffred** [comment]
@keenanjohnson, How is the clips design holding up?  Is it working out ok during printing and assembly?  I still need to print an entire assembly.  I'll give it a shot with PrusaSlicer

**Reply by keenanjohnson** [reply]
Things seem to be doing great so far! I have put together 4 with this design with no real issues!
